common/src/main/java/com/wynntils/models/items/encoding/impl/block/DefenseDataTransformer.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.items.encoding.impl.block;
+
+import com.wynntils.models.elements.type.Element;
+import com.wynntils.models.items.encoding.data.DefenseData;
+import com.wynntils.models.items.encoding.type.DataTransformer;
+import com.wynntils.models.items.encoding.type.ItemTransformingVersion;
+import com.wynntils.utils.UnsignedByteUtils;
+import com.wynntils.utils.type.ArrayReader;
+import com.wynntils.utils.type.ErrorOr;
+import com.wynntils.utils.type.Pair;
+import com.wynntils.utils.type.UnsignedByte;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefenseDataTransformer extends DataTransformer<DefenseData> {
+    @Override
+    protected ErrorOr<UnsignedByte[]> encodeData(ItemTransformingVersion version, DefenseData data) {
+        return switch (version) {
+            case VERSION_1 -> encodeDefenseData(data);
+        };
+    }
+
+    @Override
+    public ErrorOr<DefenseData> decodeData(ItemTransformingVersion version, ArrayReader<UnsignedByte> byteReader) {
+        return switch (version) {
+            case VERSION_1 -> decodeDefenseData(byteReader);
+        };
+    }
+
+    @Override
+    public byte getId() {
+        return 11;
```

Comment:
Another idea is to have a "global address enum", which contains the ID for all these data types. One upshot is that you avoid the "magic number in code" problem, and at the same time you get a overview of existing ids, and what the next free number would be. It also feels a bit contrary to the free-standing design currently used, but then again, a central rule coordinating numbers so they don't collide is necessary, so maybe it's better to actually respresent it as a enum class.

Reply:
In fact, the more I think about it, the better I like this idea. Don't restore the old ID fields, create a enum instead with the responsibility of owning the ID numbers.